src/gui/sdlmain.cpp

```diff
@@ -4533,27 +4544,46 @@ void restart_program(std::vector<std::string> & parameters) {
 		newargs[i] = parameters[i].data();
 	}
 	newargs[parameters.size()] = nullptr;
+#endif // WIN32
+
 	MIXER_CloseAudioDevice();
 	Delay(50);
 	QuitSDL();
+
 #if C_DEBUG
 	// shutdown curses
 	DEBUG_ShutDown(nullptr);
-#endif
+#endif // C_DEBUG
 
-	if(execvp(newargs[0], newargs) == -1) {
 #ifdef WIN32
-		if(newargs[0][0] == '\"') {
-			//everything specifies quotes around it if it contains a space, however my system disagrees
-			std::string edit = parameters[0];
-			edit.erase(0,1);edit.erase(edit.length() - 1,1);
-			//However keep the first argument of the passed argv (newargs) with quotes, as else repeated restarts go wrong.
-			if(execvp(edit.c_str(), newargs) == -1) E_Exit("Restarting failed");
-		}
-#endif
+	// Input structure with a bunch of stuff we probably don't care about.
+	// Zero it out to use defaults save for the size field.
+	STARTUPINFO startup_info = {};
+	startup_info.cb = sizeof(startup_info);
+
+	// Output strcture we also don't care about.
+	PROCESS_INFORMATION process_information = {};
+
+	if (!CreateProcess(
+			nullptr, // Filename - nullptr to parse from command line
+			const_cast<char *>(command_line.c_str()), // Filename + arguments as a single C string
+			nullptr, // lpProcessAttributes - nullptr for default
+			nullptr, // lpThreadAttributes  - nullptr for default
+			FALSE, // Do not inherit handles
+			DETACHED_PROCESS, // Creation flags. DETACHED_PROCESS fixes https://github.com/dosbox-staging/dosbox-staging/issues/3346
+			nullptr, // Environment variables - nullptr to use parent's enviornment
+			nullptr, // Current Directory - nullptr to use parent's current directory
+			&startup_info,
+			&process_information)) {
+		E_Exit("Restarting failed");
+	}
+	ExitProcess(ERROR_SUCCESS);
```

Comment:
Wouldn't it be better to extract these into vars and put line comments above them? I really dislike inline comments, especially when they're long.

E.g.:

```cpp
// nullptr for default
const LPSECURITY_ATTRIBUTES lpProcessAttributes = nullptr;

Reply:
Done.  Just had to turn on my caps lock and let my cat walk on my keyboard to get some of those Win32 type names correct.

Reply:
> Done. Just had to turn on my caps lock and let my cat walk on my keyboard to get some of those Win32 type names correct.

Hungarian notation, man. Well, I'm Hungarian, but those prefixes still read like gibberish to me 😄